numpy/lib/stride_tricks.py

```diff
@@ -35,7 +35,19 @@ def _maybe_view_as_subclass(original_array, new_array):
     return new_array
 
 
-def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
+def _actual_range(x):
+    if not isinstance(x, np.ndarray):
+        x = np.array(x)
+    head = np.sum((np.array(x.shape) - 1) *
```

Comment:
For small lists, creating arrays creates a large overhead:
```
In [27]: x = np.ones((4,5,3,8,6))

In [28]: %timeit np.sum((np.array(x.shape) - 1) * np.array([s if s > 0 else 0 for s in x.strides]))
10000 loops, best of 3: 23.7 µs per loop

In [29]: %timeit sum((sh - 1) * (s if s > 0 else 0) for sh, s in zip(x.shape, x.strides))
100000 loops, best of 3: 2.28 µs per loop
```
So, I'd replace with the example given.